lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -18,18 +18,27 @@ def initialize(options={})
       # @options = DEFAULT_OPTIONS.merge(options)
       @options = options
       @client = build_client(@options)
+      # mutex to prevent requests and sniffing to access the
+      # connection pool at the same time
+      @sniff_request_mutex = Mutex.new
```

Comment:
@jsvd minor nitpick, but why not just call it `request_mutex`? I was actually confused when I read the code below as to why you were using a sniff_mutex to lock template update request